File: src/main/java/io/github/svegon/capi/event/ListenerList.java
package io.github.svegon.capi.event;

import java.util.List;
import java.util.function.Function;

public class ListenerList<L> extends ListenerCollection<L, List<L>> {
    ListenerList(List<L> listeners, Function<List<L>, L> invokerFactory) {
        super(listeners, invokerFactory);
    }
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/listener_collections/ListenerList.kt
package io.github.svegon.mclientapi.event.listener_collections

import java.util.Collections

class ListenerList<L> internal constructor(listeners: MutableList<L>, invokerFactory: (List<L>) -> L) :
    ListenerCollection<L, MutableList<L>, List<L>>(listeners, invokerFactory) {
    override fun makeUnmodifiable(listeners: MutableList<L>): List<L> {
        return Collections.unmodifiableList(listeners)
    }
}
